code/modules/experisci/experiment/types/experiment.dm

```diff
@@ -95,11 +95,30 @@
 /**
  * Called when you complete an experiment, makes sure the techwebs knows the experiment was finished, and tells everyone it happend, yay!
  */
-/datum/experiment/proc/finish_experiment(datum/component/experiment_handler/experiment_handler)
+/datum/experiment/proc/finish_experiment(datum/component/experiment_handler/experiment_handler, datum/techweb/linked_web_override)
 	completed = TRUE
-	experiment_handler.selected_experiment = null
-	var/announcetext = experiment_handler.linked_web.complete_experiment(src)
-	experiment_handler.announce_message_to_all(announcetext)
+	if(!experiment_handler && !linked_web_override)
+		CRASH("finish_experiment() called without either experiment_handler or linked_web_override being set")
+	experiment_handler?.selected_experiment = null
+	var/datum/techweb/linked_web = linked_web_override || experiment_handler.linked_web
+	var/announcetext = linked_web.complete_experiment(src)
+	announce_message_to_all(announcetext, linked_web)
+
+/**
+ * Announces a message to all experiment handlers
+ *
+ * Arguments:
+ * * message - The message to announce
+ * * linked_web - the linked techweb we want to target. Prevent experiment handlers not linked to said techweb from receiving the message
+ */
+/datum/experiment/proc/announce_message_to_all(message, datum/techweb/linked_web)
+	for(var/datum/component/experiment_handler/experi_handler as anything in GLOB.experiment_handlers)
+		if(experi_handler.linked_web != linked_web)
+			continue
+		if((experi_handler.config_flags & EXPERIMENT_CONFIG_ALWAYS_ANNOUNCE) && !experi_handler.is_compatible_experiment(src))
+			continue
+		var/atom/movable/experi_parent = experi_handler.parent
+		experi_parent.say(message)
```

Comment:
earlier we define
> /// Experiment handlers with this flag will announce when an experiment is completed even if it isn't compatible with them.
#define EXPERIMENT_CONFIG_ALWAYS_ANNOUNCE (1 << 4)

but this looks like it specifically *doesn't* announce it if `EXPERIMENT_CONFIG_ALWAYS_ANNOUNCE` is set and the experiment isn't compatible?